pinot-core/src/main/java/org/apache/pinot/core/geospatial/transform/function/BaseBinaryGeoTransformFunction.java

```diff
@@ -0,0 +1,144 @@
+    Preconditions.checkArgument(transformFunction.getResultMetadata().getDataType() == FieldSpec.DataType.BYTES
+            || transformFunction instanceof LiteralTransformFunction,
+        "The second argument must be of type BYTES , but was %s",
+            transformFunction.getResultMetadata().getDataType()
+        );
+    if (transformFunction instanceof LiteralTransformFunction) {
+      _secondLiteral = GeometrySerializer.deserialize(
+          BytesUtils.toBytes(((LiteralTransformFunction) transformFunction).getLiteral()));
+    } else {
+      _secondArgument = transformFunction;
+    }
+  }
+
+  @Override
+  public TransformResultMetadata getResultMetadata() {
+    return INT_SV_NO_DICTIONARY_METADATA;
+  }
+
+  @Override
+  public int[] transformToIntValuesSV(ProjectionBlock projectionBlock) {
+    if (_intResults == null) {
+      _intResults = new int[DocIdSetPlanNode.MAX_DOC_PER_CALL];
+    }
+    final byte[][] firstValues;
+    final byte[][] secondValues;
+    if (_firstArgument != null) {
+      firstValues = _firstArgument.transformToBytesValuesSV(projectionBlock);
+    } else {
+      firstValues = null;
+    }
+    if (_secondArgument != null) {
+      secondValues = _secondArgument.transformToBytesValuesSV(projectionBlock);
+    } else {
+      secondValues = null;
+    }
+    for (int i = 0; i < projectionBlock.getNumDocs(); i++) {
+      Geometry firstGeometry = firstValues == null ? _firstLiteral : GeometrySerializer.deserialize(firstValues[i]);
+      Geometry secondGeometry = secondValues == null ? _secondLiteral : GeometrySerializer.deserialize(secondValues[i]);
+      _intResults[i] = transformGeometryToInt(firstGeometry, secondGeometry);
+    }
+    return _intResults;
+  }
+
+  @Override
+  public double[] transformToDoubleValuesSV(ProjectionBlock projectionBlock) {
+    if (_doubleResults == null) {
+      _doubleResults = new double[DocIdSetPlanNode.MAX_DOC_PER_CALL];
+    }
+    final byte[][] firstValues;
+    final byte[][] secondValues;
+    if (_firstArgument != null) {
+      firstValues = _firstArgument.transformToBytesValuesSV(projectionBlock);
+    } else {
+      firstValues = null;
+    }
+    if (_secondArgument != null) {
+      secondValues = _secondArgument.transformToBytesValuesSV(projectionBlock);
+    } else {
+      secondValues = null;
+    }
+    for (int i = 0; i < projectionBlock.getNumDocs(); i++) {
+      Geometry firstGeometry = firstValues == null ? _firstLiteral : GeometrySerializer.deserialize(firstValues[i]);
+      Geometry secondGeometry = secondValues == null ? _secondLiteral : GeometrySerializer.deserialize(secondValues[i]);
+      _doubleResults[i] = transformGeometryToDouble(firstGeometry, secondGeometry);
+    }
+    return _doubleResults;
+  }
+
+  public int transformGeometryToInt(Geometry firstGeometry, Geometry secondGeometry) {
+    throw new UnsupportedOperationException("Unsupported!");
```

Comment:
(minor) No need to put message for `UnsupportedOperationException`
```suggestion
    throw new UnsupportedOperationException();
```